knowage-core/src/main/java/it/eng/spagobi/engines/drivers/chart/ChartDriver.java

```diff
@@ -224,6 +226,23 @@ private Map applyService(Map parameters, BIObject biObject) {
 		return parameters;
 	}
 
+	private static DocumentBuilderFactory getFactory() {
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		try {
+			dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+			dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+			dbf.setFeature("http://xml.org/sax/features/external-parameterentities", false);
+			dbf.setFeature("http://xml.org/sax/features/external-generalentities", false);
+		} catch (ParserConfigurationException e) {
+			logger.error("Error loading XML document: " + e.getMessage(), e);
```

Comment:
The error message is misleading: we are not loading XML document in that block

Reply:
Fixed